koku/api/report/all/openshift/provider_map.py

```diff
@@ -28,13 +28,50 @@
 from api.models import Provider
 from api.report.provider_map import ProviderMap
 from reporting.models import OCPAllCostLineItemDailySummary
+from reporting.models import OCPAllCostLineItemDailySummaryCompute
+from reporting.models import OCPAllCostLineItemDailySummaryDatabase
+from reporting.models import OCPAllCostLineItemDailySummaryNetwork
+from reporting.models import OCPAllCostLineItemDailySummaryStorage
 from reporting.models import OCPAllCostLineItemProjectDailySummary
+from reporting.models import OCPAllCostLineItemProjectDailySummaryCompute
+from reporting.models import OCPAllCostLineItemProjectDailySummaryDatabase
+from reporting.models import OCPAllCostLineItemProjectDailySummaryNetwork
+from reporting.models import OCPAllCostLineItemProjectDailySummaryStorage
+
+
+DEFAULT_SUMMARY = (OCPAllCostLineItemDailySummary, OCPAllCostLineItemProjectDailySummary)
+STORAGE_SUMMARY = (OCPAllCostLineItemDailySummaryStorage, OCPAllCostLineItemProjectDailySummaryStorage)
+COMPUTE_SUMMARY = (OCPAllCostLineItemDailySummaryCompute, OCPAllCostLineItemProjectDailySummaryCompute)
+NETWORK_SUMMARY = (OCPAllCostLineItemDailySummaryNetwork, OCPAllCostLineItemProjectDailySummaryNetwork)
+DATABASE_SUMMARY = (OCPAllCostLineItemDailySummaryDatabase, OCPAllCostLineItemProjectDailySummaryDatabase)
+MODEL_MAP = {
+    ("storage", None): STORAGE_SUMMARY,
+    ("instance_type", None): COMPUTE_SUMMARY,
+    ("costs", "network"): NETWORK_SUMMARY,
+    ("costs", "database"): DATABASE_SUMMARY,
+    ("storage_by_project", None): STORAGE_SUMMARY,
+    ("instance_type_by_project", None): STORAGE_SUMMARY,
+    ("costs_by_project", "network"): NETWORK_SUMMARY,
+    ("costs_by_project", "database"): DATABASE_SUMMARY,
+}
+
+
+def resolver(report_type, report_subtype=None):
```

Comment:
These are fine, but just a different implementation than what was referenced in the PRs in the issue. So we should probably decide now which route we want to go down for the other issues and either change this or the existing AWS implementation.